specs/Deployment/DeploymentAPI.md

```diff
@@ -0,0 +1,133 @@
+the WinAppSDK dependencies are not satisfied, and if that is the case, provide a means to resolve the
+missing dependency.
+
+The key scenario that is expected to have a missing dependency is for packaged MSIX which have
+declared the WinAppSDK frameworks as a dependency. In this scenario it is likely the main packages
+would be missing because they cannot declare a dependency on main packages and it will not be
+deployed by the Store or other mechanisms which only expect a single main package. For MSIX
+packages with permissions, this dependency could be immediately resolved online or offline if the
+package size is small and the framework carries the dependency.
+
+A class of MSIX packages that are Full Trust or have the packageManagement capability will have
+the permissions to immediately resolve the dependency. These packages are the target for this
+API. These packages will have the capability to deploy packages and satisfy the dependency.
+Since the main packages for the WinAppSDK are expected to be very small, this API has a
+streamlined solution for this scenario, which is to carry the WinAppSDK main packages inside
+the WinAppSDK frameworks, and leverage the full trust capability of the caller to deploy the main
+packages.
+
+
+# Description
+
+When a packaged app using the WinAppSDK is deployed from the Store, it is expected to only have
+the application's main package and frameworks, and the WinAppSDK other main packages may not be
+present, or may not be at the required version. This API enables a developer to detect when the
+WinAppSDK is incomplete and take corrective action.
+
+Since this API is part of the WinAppSDK framework, the framework itself must be present.
+For MSIX packages, the framework must be listed as a dependency in the application manifest and
+will not be deployable without a min version being present. 
+
+Developers must not assume that the WinAppSDK main package is present, or that the version
+present is the minimum version required to support the feature set they expect. Instead, this API
+should be used to validate the WinAppSDK is the correct version and all required packages are present.
+
+The developer scenarios for this API is to answer the questions:
+1. Does this system have the min version of the Windows App SDK I require and are they usable?
+2. What do I do to resolve this if it doesn't?
+
+The API addresses #1 by a simple query of the version they seek and a possible check for OS
+updates. #2 has two possible answers. First is direct resolution by the developer by acquiring and
+installing the packages if possible. Since the main packages are small and the target developer
+scenario will have permissions to deploy them, we will minimize developer and user pain by
+including the main package MSIX packages within the framework, and this API will allow for the
+deployment of that package if needed.
+
+Processes lacking permission to deploy MSIX are out-of-scope at this time.
+
+# Examples
```

Comment:
Yes, that is the intent. Conceptually it's the same as initializing COM. The reason it is not built into the APIs themselves is because it may affect your usage, and deployment of packages is not instantaneous. It may take a few seconds to install the Main/Singleton the first time, and because of that we wanted to give you control over when that could happen.

The APIs that require that should also return a meaningful result if the expected runtime packages are not present and they fail for that reason.